package/MDAnalysis/analysis/nuclinfo.py

```diff
@@ -336,24 +364,45 @@ def tors(universe, seg, i):
-                             " atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i))
-    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i), " atom {0!s} {1!s} C3\' ".format(seg, i),
-                             " atom {0!s} {1!s} O3\' ".format(seg, i), " atom {0!s} {1!s} P    ".format(seg, i + 1))
-    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i),
-                             " atom {0!s} {1!s} P    ".format(seg, i + 1), " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
+    a = universe.select_atoms(" atom {0!s} {1!s} O3\' ".format(seg, i - 1),
+                              " atom {0!s} {1!s} P  ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i))
+
+    b = universe.select_atoms(" atom {0!s} {1!s} P    ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i))
+
+    g = universe.select_atoms(" atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i))
+
+    d = universe.select_atoms(" atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i))
+
+    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1))
+
+    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
     try:
-        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i), " atom {0!s} {1!s} C1\' ".format(seg, i),
-                                 " atom {0!s} {1!s} N1 ".format(seg, i), " atom {0!s} {1!s} C2  ".format(seg, i))
+        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i),
+                                  " atom {0!s} {1!s} C1\' ".format(seg, i),
+                                  " atom {0!s} {1!s} N1 ".format(seg, i),
+                                  " atom {0!s} {1!s} C2  ".format(seg, i))
     except:
```

Comment:
I am not sure an empty selection should be an error. I think what should be done in that specific piece of code is to check the length of the selection rather than using try/except.